sylvia/tests/reply_data.rs

```diff
@@ -0,0 +1,287 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::to_json_binary;
+use cw_storage_plus::Item;
+use cw_utils::{MsgInstantiateContractResponse, ParseReplyError};
+use noop_contract::sv::{Executor, NoopContractInstantiateBuilder};
+use sv::SubMsgMethods;
+use sylvia::builder::instantiate::InstantiateBuilder;
+use sylvia::ctx::{ExecCtx, InstantiateCtx, ReplyCtx};
+use sylvia::cw_std::{Addr, Binary, Response, StdError, SubMsg};
+use sylvia::types::Remote;
+use sylvia::{contract, entry_points};
+use thiserror::Error;
+
+#[allow(dead_code)]
+mod noop_contract {
+    use cosmwasm_std::{Binary, StdResult};
+    use sylvia::ctx::{ExecCtx, InstantiateCtx};
+    use sylvia::{contract, entry_points};
+
+    use sylvia::cw_std::Response;
+
+    pub struct NoopContract;
+
+    #[entry_points]
+    #[contract]
+    impl NoopContract {
+        pub const fn new() -> Self {
+            Self
+        }
+
+        #[sv::msg(instantiate)]
+        fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[sv::msg(exec)]
+        fn noop(&self, _ctx: ExecCtx, data: Option<Binary>) -> StdResult<Response> {
+            let resp = match data {
+                Some(data) => Response::new().set_data(data),
+                None => Response::new(),
+            };
+
+            Ok(resp)
+        }
+    }
+}
+
+#[cw_serde]
+pub struct InstantiatePayload {
+    pub sender: Addr,
+}
+
+#[derive(Error, Debug, PartialEq)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("{0}")]
+    ParseReply(#[from] ParseReplyError),
+}
+
+pub struct Contract {
+    remote: Item<Remote<'static, noop_contract::NoopContract>>,
+}
+
+#[entry_points]
+#[contract]
+#[sv::error(ContractError)]
+#[sv::features(replies)]
+impl Contract {
+    pub fn new() -> Self {
+        Self {
+            remote: Item::new("remote"),
+        }
+    }
+
+    #[sv::msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        ctx: InstantiateCtx,
+        remote_code_id: u64,
+    ) -> Result<Response, ContractError> {
+        // Custom type can be used as a payload.
```

Comment:
Let's have this information somewhere in the docs

Reply:
Ditto